sass `UrlRebasingImporter` rewrites urls in inline assets

### Command

build, serve

### Is this a regression?

- [x] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

18.0.2

### Description

Given these styles files:
```scss
// app.component.scss
@use './foo.css';

// foo.css
.x{background:url("data:image/svg+xml") 0 0 no-repeat,url("data:image/svg+xml;clip-path='url(%23a)'") 0 0 no-repeat;}
```

`ng build` throws this error:
```log
❯ ng build
Application bundle generation failed. [2.303 seconds]

✘ [ERROR] expected ")".
  ╷
1 │ .x{background:url("data:image/svg+xml") 0 0 no-repeat,url("data:image/svg+xml;clip-path='url("||file:%23a")'") 0 0 no-repeat;}
  │                                                                                               ^
  ╵
  src/app/part-of-adyen.css 1:95  @use
  src/app/app.component.scss 1:1  root stylesheet [plugin angular-sass]
```



### Minimal Reproduction

https://github.com/sod/issue-css
```sh
git clone git@github.com:sod/issue-css.git
cd issue-css
yarn
ng build
```

You can also try the original css where I distilled the css from:
![CleanShot 2024-07-19 at 15 08 43@2x](https://github.com/user-attachments/assets/1de0fd51-3810-492b-95b3-e5824bdcd3d5)

### Your Environment

```text
     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/


Angular CLI: 18.1.1
Node: 20.11.1
Package Manager: yarn 1.22.19
OS: darwin arm64

Angular: 18.1.1
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1801.1
@angular-devkit/build-angular   18.1.1
@angular-devkit/core            18.1.1
@angular-devkit/schematics      18.1.1
@schematics/angular             18.1.1
rxjs                            7.8.1
typescript                      5.5.3
zone.js                         0.14.8
```

---

Downgrading to angular 18.0.2 fixes the issue:

```sh
yarn set version berry
yarn config set nodeLinker node-modules
yarn
yarn up '@angular/*@18.0.2' '@angular-devkit/*@18.0.2' 'typescript@5.4'
ng build
```

![CleanShot 2024-07-19 at 15 22 49@2x](https://github.com/user-attachments/assets/3f1b69bc-692f-4ee4-ab3c-1111e85fa7b3)

Upgrading to 18.0.3 lets it reappear:
```sh
yarn up '@angular/*@18.0.3' '@angular-devkit/*@18.0.3' 'typescript@5.4' && ng build
```
![CleanShot 2024-07-19 at 15 23 39@2x](https://github.com/user-attachments/assets/208802cc-7daa-48c4-9055-cb2820b29916)

--- a/packages/angular/build/src/tools/sass/lexer.ts
+++ b/packages/angular/build/src/tools/sass/lexer.ts
@@ -55,7 +55,7 @@ export function* findUrls(
     if (pos > 0) {
       pos -= 2;
       next();
-      if (!isWhitespace(current) && current !== 0x0027 && current !== 0x003a) {
+      if (!isWhitespace(current) && current !== 0x002c && current !== 0x003a) {
         // Skip - not a url token
         pos += 3;
         continue;

--- a/packages/angular/build/src/tools/sass/lexer_spec.ts
+++ b/packages/angular/build/src/tools/sass/lexer_spec.ts
@@ -1,0 +1,31 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import { findUrls } from './lexer';
+
+describe('Sass URL lexer', () => {
+  it('should find url immediately after a colon, comma, or whitespace', () => {
+    const input = `.c{property:url("first"),url("second"), url("third");}`;
+
+    const result = [...findUrls(input)];
+
+    expect(result).toEqual([
+      { start: 16, end: 23, value: 'first' },
+      { start: 29, end: 37, value: 'second' },
+      { start: 44, end: 51, value: 'third' },
+    ]);
+  });
+
+  it('should find full url with string containing url function', () => {
+    const input = `.c{property:url("url('abc')");}`;
+
+    const result = [...findUrls(input)];
+
+    expect(result).toEqual([{ start: 16, end: 28, value: `url('abc')` }]);
+  });
+});

